pkg/kube/inject/inject.go

```diff
@@ -572,6 +572,29 @@ func InjectionData(sidecarTemplate, valuesConfig, version string, typeMetadata *
 		"toLower":             strings.ToLower,
 	}
 
+	// Migrating from values.yaml to mesh or env (for istiod - set using fromEnv instead of values)
+	// This allows an optional config map with vendor-specific settings/overrides - alternative is creating
+	// vendor-specific injection templates or a huge mess.
+	funcMap["value"] = func(key string, def string) string {
+		fromEnv := os.Getenv(strings.ReplaceAll(key, ".", "_"))
+		if fromEnv != "" {
+			return fromEnv
+		}
+		// Special case for 'Values.global' - where most settings are
+		if !strings.Contains(key, ".") {
+			global := values["global"]
+			if globalM, ok := global.(map[string]interface{}); ok {
+				sval := globalM[key]
+				if sval != nil {
+					if val, ok := sval.(string); ok {
```

Comment:
Why only support string?

Reply:
This is not intended as a generic jsonpatch reinvention - just a way to not have to keep values.yaml around for injector. I'll add whatever is needed - so far I replaced only strings.

Reply:
We don't need to reinvent anything with `{{ valueOrDefault (getEnv "PILOT_FOO_BAR") .Values.global.foo.bar }}` other than added the env function though?

Reply:
Well - the point is to stop using .Values.

But it doesn't matter - since the api will be 'env' map in proxy config, that's the way to go.

This PR is just a stop gap until api is agreed.